libs/remote_storage/src/s3_bucket.rs

```diff
@@ -353,10 +354,17 @@ impl RemoteStorage for S3Bucket {
 
     /// See the doc for `RemoteStorage::list_files`
     async fn list_files(&self, folder: Option<&RemotePath>) -> anyhow::Result<Vec<RemotePath>> {
-        let folder_name = folder
+        let mut folder_name = folder
             .map(|p| self.relative_path_to_s3_object(p))
             .or_else(|| self.prefix_in_bucket.clone());
 
```

Comment:
this change might be unnecessary now after the refactor to `relative_path_to_s3_object`

Reply:
yup it is no longer necessary. i removed it